packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -298,14 +306,42 @@ export class PendingStateManager implements IDisposable {
 	 * that the batch information is correct.
 	 * @param batch - The batch that is being processed.
 	 * @param batchStartCsn - The clientSequenceNumber of the start of this message's batch
+	 * @param sequenceNumber - The sequence number of the batch message.
 	 */
 	public processPendingLocalBatch(
 		batch: InboundSequencedContainerRuntimeMessage[],
 		batchStartCsn: number,
+		sequenceNumber?: number,
 	): {
 		message: InboundSequencedContainerRuntimeMessage;
 		localOpMetadata: unknown;
 	}[] {
+		if (batch.length === 0) {
+			const pendingMessage = this.pendingMessages.peekFront();
```

Comment:
Suggestion: Move the body of this `if` block to a function `processPendingLocalEmptyBatch`